dll/win32/shell32/wine/pidl.c

```diff
@@ -485,6 +485,37 @@ LPITEMIDLIST WINAPI ILGlobalClone(LPCITEMIDLIST pidl)
     return newpidl;
 }
 
+#ifdef __REACTOS__
+static inline LPITEMIDLIST _ILUnsafeNext(LPCITEMIDLIST pidl)
+{
+    return (LPITEMIDLIST)(((BYTE*)pidl) + pidl->mkid.cb);
+}
+
+UINT _ILGetDepth(LPCITEMIDLIST pidl)
+{
+    for (UINT i = 0;; ++i)
+    {
+        if (!pidl || !pidl->mkid.cb)
+            return i;
+        pidl = _ILUnsafeNext(pidl);
+    }
+}
+
+static BOOL _ILMemCmpEqualIDList(LPCITEMIDLIST p1, LPCITEMIDLIST p2)
+{
+    for (;; p1 = _ILUnsafeNext(p1), p2 = _ILUnsafeNext(p2))
+    {
+        DWORD cb1 = p1 ? p1->mkid.cb : 0x80000000; /* Empty != NULL */
+        DWORD cb2 = p2 ? p2->mkid.cb : 0x80000000;
+        if (cb1 != cb2)
+            return FALSE;
+        if (LOWORD(cb1) == 0)
+            return cb1 == cb2;
+        if (memcmp(p1, p2, cb1))
+            return FALSE;
+    }
+}
+#else /* __REACTOS__ */
 BOOL _ILHACKCompareSimpleIds(LPCITEMIDLIST pidltemp1, LPCITEMIDLIST pidltemp2)
```

Comment:
ILHACK lol

Reply:
And you are surprised IL comparison is sometimes broken?